extension/basicauthextension/extension.go

```diff
@@ -38,50 +39,54 @@ var (
 )
 
 type basicAuth struct {
-	htpasswd  HtpasswdSettings
-	matchFunc func(username, password string) bool
+	htpasswd     HtpasswdSettings
+	matchFunc    func(username, password string) bool
+	userPassPair string
 }
 
-func newExtension(cfg *Config) (configauth.ServerAuthenticator, error) {
+func newExtension(cfg *Config) (*basicAuth, error) {
 	if cfg.Htpasswd.File == "" && cfg.Htpasswd.Inline == "" {
 		return nil, errNoCredentialSource
 	}
 	ba := basicAuth{
 		htpasswd: cfg.Htpasswd,
 	}
-	return configauth.NewServerAuthenticator(configauth.WithStart(ba.start), configauth.WithAuthenticate(ba.authenticate)), nil
+	return &ba, nil
 }
 
-func (ba *basicAuth) start(ctx context.Context, host component.Host) error {
-	var rs []io.Reader
+func (ba *basicAuth) Start(ctx context.Context, host component.Host) error {
+	var buff bytes.Buffer
 
 	if ba.htpasswd.File != "" {
-		f, err := os.Open(ba.htpasswd.File)
+		bytes, err := ioutil.ReadFile(ba.htpasswd.File)
 		if err != nil {
-			return fmt.Errorf("open htpasswd file: %w", err)
+			return fmt.Errorf("open file error: %w", err)
 		}
-		defer f.Close()
-
-		rs = append(rs, f)
-		rs = append(rs, strings.NewReader("\n"))
+		buff.Write(bytes)
+		buff.WriteString("\n")
 	}
 
 	// Ensure that the inline content is read the last.
 	// This way the inline content will override the content from file.
-	rs = append(rs, strings.NewReader(ba.htpasswd.Inline))
-	mr := io.MultiReader(rs...)
+	if len(ba.htpasswd.Inline) > 0 {
+		buff.Truncate(buff.Len())
+		buff.WriteString(ba.htpasswd.Inline)
+	}
 
-	htp, err := htpasswd.NewFromReader(mr, htpasswd.DefaultSystems, nil)
+	htp, err := htpasswd.NewFromReader(bytes.NewBuffer(buff.Bytes()), htpasswd.DefaultSystems, nil)
 	if err != nil {
 		return fmt.Errorf("read htpasswd content: %w", err)
 	}
-
+	ba.userPassPair = buff.String()
```

Comment:
Am I right to assume that this is getting the username and password from the htpasswd file? If so, that's not right.